File: packages/ios/Sources/CivUI/CivDisclosure.swift
// CivUI — CivDisclosure for SwiftUI
// Inline expandable content (mirrors the web civ-disclosure component).
//
// Placeholder body — the prop surface satisfies schema parity. The
// implementation should wire SwiftUI's DisclosureGroup with the
// label affordance. See audit-debt.md for status.

import SwiftUI

/// Accessible inline disclosure widget for "Why we ask?" justifications
/// and other supplementary content.
public struct CivDisclosure: View {
    /// Trigger text displayed beside the caret.
    public var label: String

    /// Whether the disclosure is currently open.
    @Binding public var open: Bool

    /// Called when the open/closed state changes.
    public var onToggle: ((Bool) -> Void)?

    /// Called for analytics tracking.
    public var onAnalytics: ((String, [String: Any]?) -> Void)?

    public init(
        label: String = "",
        open: Binding<Bool> = .constant(false),
        onToggle: ((Bool) -> Void)? = nil,
        onAnalytics: ((String, [String: Any]?) -> Void)? = nil
    ) {
        self.label = label
        self._open = open
        self.onToggle = onToggle
        self.onAnalytics = onAnalytics
    }

    public var body: some View {
        EmptyView()
    }
}

File: packages/android/src/main/kotlin/gov/civui/components/CivDisclosure.kt
// CivUI — CivDisclosure for Jetpack Compose
// Inline expandable content (mirrors the web civ-disclosure component).
//
// Placeholder body — the prop surface satisfies schema parity. The
// implementation should wire a Compose AnimatedVisibility / expandable
// card with the label/icon/size affordances. See audit-debt.md.

package gov.civui.components

import androidx.compose.foundation.layout.Column
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier

/**
 * Accessible inline disclosure widget for "Why we ask?" justifications
 * and other supplementary content.
 */
@Composable
fun CivDisclosure(
    label: String = "",
    open: Boolean = false,
    icon: String = "chevron-right",
    size: String = "default",
    modifier: Modifier = Modifier,
    onToggle: ((Boolean) -> Unit)? = null,
    onAnalytics: ((event: String, data: Map<String, Any>?) -> Unit)? = null,
    content: @Composable () -> Unit = {},
) {
    Column(modifier = modifier) {
        // Placeholder — see audit-debt.md for native implementation status.
    }
}
